### PR TITLE
Fix missing ParameterModule export in GPT-OSS compiler path

### PR DESCRIPTION
**Pull request by danielhanchen**
## Summary
Fixes GPT-OSS compiled-cache failures where generated modules can reference `ParameterModule` but the symbol is not available from `transformers.models.gpt_oss.modeling_gpt_oss`.

## Root cause
`ParameterModule` is defined in `unsloth_zoo.temporary_patches.gpt_oss`, but compiler-generated code expects symbols to be resolvable from the GPT-OSS modeling namespace.

## Changes
- Added `patch_gpt_oss_compiler_exports()` in `unsloth_zoo/temporary_patches/gpt_oss.py`.
- Exported compiler-required symbols into `transformers.models.gpt_oss.modeling_gpt_oss`:
  - `ParameterModule`
  - `swiglu_torch_forward`
  - `dtype_from_config`
  - `transformers_version`
  - `Version`
- Registered the hook via `TEMPORARY_PATCHES.append(patch_gpt_oss_compiler_exports)`.
- Normalized GPT-OSS guard matching with `UNSLOTH_MODEL_NAME.replace("-", "_")` usage across key 4-bit/load guards.
- Cleaned compiler export patch flow in exception path and removed redundant `pass`.
- Updated compiler decorator stripping to handle:
  - `@use_kernel_forward_from_hub(...)`
  - `@auto_docstring`
  This removes noisy unknown-decorator warnings in GPT-OSS compile paths.

## Validation
Run dirs:
- `logs/pr519_verify_20260226_031808/test_gpt_oss_compiler_export.log`
- `logs/pr519_verify_20260226_031808/test_gpt_oss_compiler_export_after_autodoc_reinstall.log`

Results:
- GPT-OSS compiler export smoke test: PASS
- `@use_kernel_forward_from_hub` unknown decorator warning: removed
- `@auto_docstring` unknown decorator warning: removed

## Notes
This change is scoped to symbol export and compile-time compatibility. It does not modify model math.

### CODE REVIEW

File: unsloth_zoo/temporary_patches/gpt_oss.py [hunk 1]
```diff
@@ -648,6 +648,26 @@ def _load_from_state_dict(
         )
 
 
+def patch_gpt_oss_compiler_exports():
+    model_name = os.environ.get("UNSLOTH_MODEL_NAME", "").replace("-", "_")
```
Comment (Datta0) [inline unsloth_zoo/temporary_patches/gpt_oss.py:652]: Ideally, if our temporary_patches creates and references a class, compiler.py should automatically handle copying it over. But for now this is fine.

File: unsloth_zoo/temporary_patches/gpt_oss.py [hunk 2]
```diff
@@ -648,6 +648,26 @@ def _load_from_state_dict(
...
+    if "gpt_oss" not in model_name:
+        return
+    try:
+        import transformers.models.gpt_oss.modeling_gpt_oss
+    except Exception as e:
+        return raise_error("transformers.models.gpt_oss.modeling_gpt_oss", e)
+
+    # Export helpers so compiler generated GPT-OSS modules can resolve symbols.
+    m = transformers.models.gpt_oss.modeling_gpt_oss
+    m.ParameterModule = ParameterModule
```
Comment (Datta0) [inline unsloth_zoo/temporary_patches/gpt_oss.py:663]: My only worry here is that this would leave `from unsloth_zoo.temporary_patches.gpt_oss import ParameterModule` in the `unsloth_compiled_module_gpt_oss.py`. Can you please verify where the import resolves to and it doesnot create a dependency cycle?

**Review by Datta0 — APPROVED** [review]

**Comment by danielhanchen** [comment]
Addressed review comments on #519 and verified import resolution:

- Added guard normalization for hyphenated GPT-OSS names across key 4-bit/load guards.
- Cleaned `patch_gpt_oss_compiler_exports` exception path and removed redundant trailing `pass`.
- Updated compiler rewrite to strip `@use_kernel_forward_from_hub(...)` and `@auto_docstring` decorators so these no longer emit unknown-decorator warnings in GPT-OSS compile path.

Datta concern check (dependency cycle):
- Ran `temp/test_gpt_oss_compiler_export.py` after reinstalling from source on this branch.
- Compiled module generated at `unsloth_compiled_cache/unsloth_compiled_module_gpt_oss.py` and import path resolved successfully.
- No dependency cycle hit; test passes.

Latest verification logs:
- `logs/pr519_verify_20260226_031808/test_gpt_oss_compiler_export_after_autodoc_reinstall.log`
